### PR TITLE
Add CONTRIBUTING.md and requirements-devel.txt. Fixes #196

### LINKED ISSUE #196: Creating guidelines for contributing
I'd like to create a `CONTRIBUTING.md` document to state out some rules for PR authors and reviewers. My initial ideas are:

- maximum line length is 100 chars (we don't have this right now, but we can fix that in time and we can only accept PRs that adhere to this from now on)
- there must be no pep8/pyflakes errors unless there's a compelling reason to break these (which is very rare)
- methods that are to be called by different parts of code (i.e. public methods and functions) must have docstrings
- code must be "readable" - meaning:
  - comments where needed
  - separating logical parts of functions by a newline
  - meaningful variable names
- landscape.io test must come out clean
- travis tests must pass
- new features must have user documentation
- new features must have tests (at least a basic smoketest)

What else? Let's brainstorm! Once we have a complete list and approval from all current comitters, I'll create the contribution guidelines and send them as a PR.

### PR DESCRIPTION
**Pull request by bkabrda**
Note: I actually omitted running pylint, since I think that it reports a whole bunch of errors not reported by landscape.io. So let's stay with pep8 and pyflakes for now.
I also added requirements-devel.txt, which contains all runtime and test dependencies as well as tools for additional checks (pyflakes, pep8 and pytest-cov).

### CODE REVIEW

**Comment by bkabrda** [comment]
Ok, I'm quite sure that coveralls failure is a false positive, because I added no code whatsoever :)

**Comment by bkabrda** [comment]
> I'd like to use LGTM (looks-good-to-me) to indicate acceptance.

So would you like to use *only* LGTM or LGTM *as well as +1 and :+1:*? I'd prefer keeping maximum two of these, since having more makes issues harder to scan for core dev consent.

**Comment by jpopelka** [comment]
LGTM only, that's what Docker people [use](https://docs.docker.com/project/review-pr/#how-we-process-your-review) and it LGTM ;-)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+## Merging Code
+
+Assuming your pull request fulfilled all the above criteria, your code is almost ready to merge. You just need to wait for one of the core devs of Atomic Reactor to give your pull request a +1 or :thumbsup: and do the actual merge. If the core dev thinks something is wrong with your pull request, he'll tell you and let you fix it. Discussion is always welcome!
+
+Current core devs are:
+* Jirka Popelka (@jpopelka)
+* Slavek Kabrda (@bkabrda)
+* Tim Waugh (@twaugh)
+* Tomas Tomecek (@TomasTomecek)
```
Comment (TomasTomecek) [inline CONTRIBUTING.md]: You forgot Martin: @mmilata

**Comment by bkabrda** [comment]
I added mmilata as a core dev and replaced +1/:+1: with LGTM.

**Comment by twaugh** [comment]
LGTM

**Comment by TomasTomecek** [comment]
ready for merge?

**Comment by bkabrda** [comment]
Yeah, I think so.

**Pull request merged**